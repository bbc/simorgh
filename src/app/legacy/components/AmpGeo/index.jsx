import React from 'react';

const configuration = {
  AmpBind: true,
  ISOCountryGroups: {
    eea: [
      'at',
      'ax',
      'be',
      'bg',
      'bl',
      'cy',
      'cz',
      'de',
      'dk',
      'ea',
      'ee',
      'es',
      'fi',
      'fr',
      'gb',
      'gf',
      'gg',
      'gi',
      'gp',
      'gr',
      'hr',
      'hu',
      'ic',
      'ie',
      'im',
      'is',
      'it',
      'je',
      'li',
      'lt',
      'lu',
      'lv',
      'mf',
      'mq',
      'mt',
      'nc',
      'nl',
      'no',
      'pf',
      'pl',
      'pm',
      'pt',
      're',
      'ro',
      'se',
      'si',
      'sj',
      'sk',
      'tf',
      'va',
      'wf',
      'yt',
    ],
    gbOrUnknown: ['gb', 'gg', 'im', 'je', 'uk', 'unknown'],
  },
};

export const AMP_GEO_SCRIPT = (
  <script
    async
    custom-element="amp-geo"
    src="https://cdn.ampproject.org/v0/amp-geo-0.1.js"
  />
);

const AmpGeo = () => (
  <amp-geo layout="nodisplay">
    <script
      type="application/json"
      // eslint-disable-next-line react/no-danger
      dangerouslySetInnerHTML={{ __html: JSON.stringify(configuration) }}
    />
  </amp-geo>
);

export default AmpGeo;
