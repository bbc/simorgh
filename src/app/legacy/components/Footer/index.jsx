import React from 'react';
import styled from '@emotion/styled';
import { arrayOf, shape, string, node, bool } from 'prop-types';
import {
  getSansBold,
  getSansRegular,
} from '#psammead/psammead-styles/src/font-styles';
import { getBrevier } from '#psammead/gel-foundations/src/typography';
import {
  GEL_SPACING,
  GEL_SPACING_DBL,
  GEL_MARGIN_BELOW_400PX,
  GEL_MARGIN_ABOVE_400PX,
} from '#psammead/gel-foundations/src/spacings';
import {
  GEL_GROUP_2_SCREEN_WIDTH_MIN,
  GEL_GROUP_4_SCREEN_WIDTH_MIN,
} from '#psammead/gel-foundations/src/breakpoints';

import { AmpCookieSettingsButton } from '#containers/ConsentBanner/Banner/cookie.amp';
import Link from './Link';
import List from './List';

const SitewideLinksWrapper = styled.div`
  ${({ script }) => script && getBrevier(script)}
  ${({ service }) => service && getSansRegular(service)}
  background-color: ${props => props.theme.palette.EBON};

  @media (max-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    padding: 0 ${GEL_MARGIN_BELOW_400PX};
  }
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    padding: 0 ${GEL_MARGIN_ABOVE_400PX};
  }
`;

const ConstrainedWrapper = styled.div`
  max-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN};
  margin: 0 auto;
  ${({ trustProjectLink }) =>
    trustProjectLink && `padding-top: ${GEL_SPACING};`}
`;

const StyledParagraph = styled.p`
  color: ${props => props.theme.palette.WHITE};
  margin: 0;
  padding: ${GEL_SPACING_DBL} 0;

  // removes padding which creates touch target from the final "inline" link so the Focus Indicator doesn't obscure other text.
  a {
    padding: 0;
  }
`;

const StyledAmpCookieSettingsButton = styled(AmpCookieSettingsButton)`
  ${({ service }) => service && getSansBold(service)}
  background: none;
  border: none;
  color: ${props => props.theme.palette.WHITE};
  cursor: pointer;
  display: block;
  padding: ${GEL_SPACING} 0 ${GEL_SPACING};
  text-decoration: none;
  text-align: left;
  width: 100%;

  &:hover,
  &:focus {
    text-decoration: underline;
  }
`;

const SitewideLinks = ({
  links,
  trustProjectLink,
  copyrightText,
  externalLink,
  isAmp,
  script,
  service,
}) => {
  const elements = links.map(({ id, text, href, lang }) => {
    if (isAmp && id === 'COOKIE_SETTINGS') {
      return (
        <StyledAmpCookieSettingsButton lang={lang} service={service}>
          {text}
        </StyledAmpCookieSettingsButton>
      );
    }
    return <Link service={service} text={text} href={href} lang={lang} />;
  });

  return (
    <SitewideLinksWrapper script={script} service={service}>
      <ConstrainedWrapper trustProjectLink={trustProjectLink}>
        <List
          service={service}
          elements={elements}
          trustProjectLink={trustProjectLink}
        />
        <StyledParagraph>
          {copyrightText}{' '}
          <Link
            service={service}
            text={externalLink.text}
            href={externalLink.href}
            inline
          />
        </StyledParagraph>
      </ConstrainedWrapper>
    </SitewideLinksWrapper>
  );
};

const linkPropTypes = shape({
  href: string.isRequired,
  text: string.isRequired,
});

SitewideLinks.propTypes = {
  links: arrayOf(linkPropTypes.isRequired).isRequired,
  copyrightText: node.isRequired,
  trustProjectLink: linkPropTypes,
  externalLink: linkPropTypes.isRequired,
  isAmp: bool,
  script: shape({}),
  service: string,
};

SitewideLinks.defaultProps = {
  script: null,
  service: null,
  isAmp: false,
  trustProjectLink: null,
};

export default SitewideLinks;
