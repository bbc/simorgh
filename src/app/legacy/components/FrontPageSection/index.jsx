import styled from '@emotion/styled';

// Components
import { GEL_GROUP_4_SCREEN_WIDTH_MIN } from '#psammead/gel-foundations/src/breakpoints';

const FrontPageSection = styled.div`
  /* To centre page layout for Group 4+ */
  margin: 0 auto;
  width: 100%; /* Needed for IE11 */
  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
    max-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN};
  }
`;

export default FrontPageSection;
