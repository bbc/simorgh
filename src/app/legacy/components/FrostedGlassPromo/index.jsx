import React, { useContext } from 'react';
import { shape, node, string, number } from 'prop-types';
import styled from '@emotion/styled';
import pick from 'ramda/src/pick';
import Lazyload from 'react-lazyload';

import { RequestContext } from '#contexts/RequestContext';
import useClickTrackerHandler from '#hooks/useClickTrackerHandler';
import { getSerifRegular } from '#psammead/psammead-styles/src/font-styles';
import { GEL_GROUP_2_SCREEN_WIDTH_MIN } from '#psammead/gel-foundations/src/breakpoints';
import {
  GEL_SPACING,
  GEL_SPACING_DBL,
} from '#psammead/gel-foundations/src/spacings';
import makeRelativeUrlPath from '#lib/utilities/makeRelativeUrlPath';
import ImageWithPlaceholder from '#containers/ImageWithPlaceholder';
import { ServiceContext } from '../../../contexts/ServiceContext';

import FrostedGlassPanel from './FrostedGlassPanel';

import withData from './withData';

const PANEL_OFFSET = 250;

const Wrapper = styled.div`
  position: relative;
  width: 100%;
  display: flex;
  flex-direction: column;
  height: 100%;
  text-decoration: none;
  &:hover {
    a {
      text-decoration: underline;
    }
  }
  &:visited {
    a {
      color: #e6e8ea;
    }
  }
`;

// This is an unfocusable element, hidden from screenreaders
// This is to allow mouse users to click anywhere on the promo
// But for keyboard and AT users, and scenarios where there is no CSS,
// only the link inside the H3 should be interactive
const ClickableArea = styled.a`
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  z-index: 5;
`;

const H3 = styled.h3`
  margin: 0;
`;

const A = styled.a`
  display: inline-block;
  ${({ service }) => service && getSerifRegular(service)}
  text-decoration: none;
  color: ${props =>
    props.isAmp ? props.theme.palette.BLACK : props.theme.palette.WHITE};
  font-size: 0.9375rem;
  line-height: 1.33;
  margin: 0.875rem ${GEL_SPACING} 0 ${GEL_SPACING};
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    font-size: 1rem;
    line-height: 1.25;
    margin: 0.875rem ${GEL_SPACING_DBL} 0 ${GEL_SPACING_DBL};
  }
  &:visited {
    color: ${props =>
      props.isAmp ? props.theme.palette.METAL : props.theme.palette.GREY_3};
  }
  &:focus {
    text-decoration: underline;
  }
`;

const LazyloadPlaceholder = styled.div`
  background-color: ${props =>
    props.isAmp ? props.theme.palette.WHITE : props.theme.palette.GREY_8};
  min-height: 100px;
  padding-bottom: ${GEL_SPACING_DBL};
`;

const LazyloadWrapper = styled(Lazyload)`
  height: 100%;
`;

const FrostedGlassPromo = ({
  image,
  children,
  footer,
  url,
  eventTrackingData,
  index,
  minimumContrast,
  paletteSize,
}) => {
  const { script, service } = useContext(ServiceContext);
  const { isAmp } = useContext(RequestContext);
  const isCanonical = !isAmp;
  const relativeUrl = makeRelativeUrlPath(url);

  const clickTracker = useClickTrackerHandler({
    ...(eventTrackingData || {}),
    url: relativeUrl,
  });

  const onClick = eventTrackingData ? clickTracker : () => {};

  const promoText = (
    <>
      <H3>
        <A
          script={script}
          service={service}
          href={relativeUrl}
          onClick={onClick}
          isAmp={isAmp}
        >
          {children}
        </A>
      </H3>
      {footer}
    </>
  );

  // The ClickableArea component is an anchor ("a") element
  // Anchors cannot be self-closing under the HTML spec
  /* eslint-disable react/self-closing-comp */
  return (
    <Wrapper data-testid={`frosted-promo-${index}`}>
      <ClickableArea
        href={relativeUrl}
        onClick={onClick}
        aria-hidden="true"
        tabIndex="-1"
      ></ClickableArea>
      <ImageWithPlaceholder
        lazyLoad
        darkMode={isCanonical}
        {...pick(
          [
            'src',
            'srcset',
            'fallbackSrcset',
            'primaryMimeType',
            'fallbackMimeType',
            'sizes',
            'alt',
            'ratio',
            'width',
            'height',
          ],
          image,
        )}
      />
      <LazyloadWrapper
        offset={PANEL_OFFSET}
        once
        placeholder={
          // Placeholder always gets rendered on AMP
          <LazyloadPlaceholder
            data-testid="frosted-glass-lazyload-placeholder"
            isAmp={isAmp}
          >
            {promoText}
          </LazyloadPlaceholder>
        }
      >
        <FrostedGlassPanel
          image={image.src}
          minimumContrast={minimumContrast}
          paletteSize={paletteSize}
        >
          {promoText}
        </FrostedGlassPanel>
      </LazyloadWrapper>
    </Wrapper>
  );
};

// This component can receive promo data in a range of different formats
// It uses a withData HoC to convert the prop to a standardised schema
// This array is the list of props that should just be passed straight through
// to the component, without requiring any preprocessing
const propsToPassThrough = ['minimumContrast', 'paletteSize', 'isAmp'];

FrostedGlassPromo.propTypes = {
  children: node.isRequired,
  url: string.isRequired,
  footer: node,
  eventTrackingData: shape({}),
  index: number,
  minimumContrast: number,
  paletteSize: number,
  image: shape({
    src: string.isRequired,
    alt: string.isRequired,
    ratio: number.isRequired,
    width: number.isRequired,
    height: number.isRequired,
    smallSrc: string,
    srcset: string,
    sizes: string,
  }).isRequired,
};

FrostedGlassPromo.defaultProps = {
  footer: null,
  eventTrackingData: null,
  index: 0,
  minimumContrast: 8,
  paletteSize: 10,
};

export default withData(FrostedGlassPromo, propsToPassThrough);
