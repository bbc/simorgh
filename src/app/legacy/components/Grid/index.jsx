import React, { useContext } from 'react';
import styled from '@emotion/styled';
import { string, node, number } from 'prop-types';
import GRID from '#psammead/psammead-grid/src';
import {
  GEL_GROUP_2_SCREEN_WIDTH_MAX,
  GEL_GROUP_3_SCREEN_WIDTH_MIN,
  GEL_GROUP_2_SCREEN_WIDTH_MIN,
  GEL_GROUP_3_SCREEN_WIDTH_MAX,
  GEL_GROUP_4_SCREEN_WIDTH_MIN,
  GEL_GROUP_4_SCREEN_WIDTH_MAX,
  GEL_GROUP_5_SCREEN_WIDTH_MIN,
} from '#psammead/gel-foundations/src/breakpoints';
import {
  GEL_MARGIN_BELOW_400PX,
  GEL_MARGIN_ABOVE_400PX,
  GEL_SPACING_DBL,
} from '#psammead/gel-foundations/src/spacings';
import { ServiceContext } from '../../../contexts/ServiceContext';

const Grid = React.forwardRef((props, ref) => {
  const { dir } = useContext(ServiceContext);

  return <GRID dir={dir} {...props} ref={ref} />;
});

const fourOfSixColumnsMaxWidthGroup4 = `30rem`;
/* (group4ColWidth 6.75rem * 4) + (3 * 16px gutters) = 27rem + 3rem = 30rem */

const eightOfTwelveColumnsMaxWidthGroup5 = `30.6rem`;
/* (group5ColWidth 2.95rem * 8) + (7 * 16px gutters) = 23.6rem + 7rem = 30.6rem */

const fiveOfSixColumnsMaxWidthScaleable = `83.33%`;
// (5 / 6) * 100 = 83.3333.. = 83.33%

const fourOfSixColumnsMaxWidthScaleable = `66.67%`;
// (4 / 6) * 100 = 66.6666.. = 66.67%

export const gelGridMargin = `
  @media (max-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    padding: 0 ${GEL_MARGIN_BELOW_400PX};
  }
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_3_SCREEN_WIDTH_MAX}) {
    padding: 0 ${GEL_MARGIN_ABOVE_400PX};
  }
`;

const layoutGridItemSmall = ({ padding = {} }) => `
  ${gelGridMargin}

  @media (max-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) {
    ${padding.group2 ? `padding: 0 ${padding.group2}` : ''};
  }
  @media (min-width: ${GEL_GROUP_2_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_2_SCREEN_WIDTH_MAX}) {
    max-width: ${fourOfSixColumnsMaxWidthScaleable};
  }
  @media (min-width: ${GEL_GROUP_3_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_3_SCREEN_WIDTH_MAX}) {
    max-width: ${fiveOfSixColumnsMaxWidthScaleable};
    ${padding.group3 ? `padding: 0 ${padding.group3}` : ''};
  }
  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_4_SCREEN_WIDTH_MAX}) {
    max-width: ${fourOfSixColumnsMaxWidthGroup4};
  }
  @media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {
    max-width: ${eightOfTwelveColumnsMaxWidthGroup5};
  }

  @supports (display: grid) {
    max-width: initial;
  }
`;

export const GelPageGrid = styled(Grid)`
  width: 100%;
  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) and (max-width: ${GEL_GROUP_4_SCREEN_WIDTH_MAX}) {
    margin: 0 auto;
    max-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN};
  }
  @media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {
    margin: 0 auto;
    max-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN};
  }
`;

/* The following components relate to Grid configuration and Grid styles used on the following page types:
 * STY,MAP,PGL,Front Page,IDX page
 */
const StyledCPSPageGrid = styled(Grid)`
  @media (min-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN}) {
    margin: 0 auto;
    max-width: ${GEL_GROUP_4_SCREEN_WIDTH_MIN};
  }
`;

export const CPSPageGrid = ({ children, ...props }) => (
  <StyledCPSPageGrid
    columns={{
      group0: 6,
      group1: 6,
      group2: 6,
      group3: 6,
      group4: 8,
      group5: 8,
    }}
    enableGelGutters
    {...props}
  >
    {children}
  </StyledCPSPageGrid>
);

export const GridWrapper = props => (
  <GelPageGrid
    {...props}
    enableGelGutters
    columns={{
      group0: 6,
      group1: 6,
      group2: 6,
      group3: 6,
      group4: 8,
      group5: 20,
    }}
  />
);

export const GridItemSmall = styled(props => (
  <Grid
    {...props}
    item
    startOffset={{
      group0: 1,
      group1: 1,
      group2: 1,
      group3: 1,
      group4: 2,
      group5: 5,
    }}
    columns={{
      group0: 6,
      group1: 6,
      group2: 4,
      group3: 5,
      group4: 4,
      group5: 8,
    }}
  />
))`
  ${layoutGridItemSmall}
`;

export const GridItemMedium = ({ gridColumnStart, gridSpan, ...props }) => (
  <Grid
    {...props}
    item
    margins={{
      group0: true,
      group1: true,
      group2: true,
      group3: true,
      group4: false,
      group5: false,
    }}
    startOffset={{
      group0: 1,
      group1: 1,
      group2: 1,
      group3: 1,
      group4: 2,
      group5: gridColumnStart,
    }}
    columns={{
      group0: 6,
      group1: 6,
      group2: 6,
      group3: 5,
      group4: 5,
      group5: gridSpan,
    }}
  />
);

export const GridItemMediumNoMargin = ({
  gridColumnStart,
  gridSpan,
  ...props
}) => (
  <Grid
    {...props}
    item
    startOffset={{
      group0: 1,
      group1: 1,
      group2: 1,
      group3: 1,
      group4: 2,
      group5: gridColumnStart,
    }}
    columns={{
      group0: 6,
      group1: 6,
      group2: 6,
      group3: 5,
      group4: 5,
      group5: gridSpan,
    }}
  />
);

export const GridItemLarge = props => (
  <Grid
    {...props}
    item
    margins={{
      group0: true,
      group1: true,
      group2: true,
      group3: true,
      group4: false,
      group5: false,
    }}
    startOffset={{
      group0: 1,
      group1: 1,
      group2: 1,
      group3: 1,
      group4: 2,
      group5: 5,
    }}
    columns={{
      group0: 6,
      group1: 6,
      group2: 6,
      group3: 6,
      group4: 6,
      group5: 12,
    }}
  />
);

export const GridItemLargeNoMargin = props => (
  <Grid
    {...props}
    item
    startOffset={{
      group0: 1,
      group1: 1,
      group2: 1,
      group3: 1,
      group4: 2,
      group5: 5,
    }}
    columns={{
      group0: 6,
      group1: 6,
      group2: 6,
      group3: 6,
      group4: 6,
      group5: 12,
    }}
  />
);

// 1.
// The max-height must be 0 at Group 5 breakpoints so that
// the item does not force the following sibling item downwards.

const PopOutAtGroup5 = styled(GridItemMedium)`
  @supports (display: grid) {
    @media (min-width: ${GEL_GROUP_5_SCREEN_WIDTH_MIN}) {
      max-height: 0; /* [1] */
      padding-top: 0.25rem;
      margin: 0 ${GEL_SPACING_DBL};
    }
  }
`;
export const PopOutGridItemMedium = ({ children, ...props }) => (
  <PopOutAtGroup5 {...props}>{children}</PopOutAtGroup5>
);

GridItemLarge.propTypes = {
  as: string,
};

GridItemLarge.defaultProps = {
  as: undefined,
};

GridItemLargeNoMargin.propTypes = {
  as: string,
};

GridItemLargeNoMargin.defaultProps = {
  as: undefined,
};

GridWrapper.propTypes = {
  as: string,
};

GridWrapper.defaultProps = {
  as: undefined,
};

GridItemMediumNoMargin.propTypes = {
  as: string,
  gridColumnStart: number,
  gridSpan: number,
};

GridItemMediumNoMargin.defaultProps = {
  as: undefined,
};

GridItemMedium.propTypes = {
  as: string,
  gridColumnStart: number,
  gridSpan: number,
};

GridItemMedium.defaultProps = {
  as: undefined,
};

GridItemMediumNoMargin.defaultProps = {
  gridColumnStart: 5,
  gridSpan: 10,
};

GridItemMedium.defaultProps = {
  gridColumnStart: 5,
  gridSpan: 10,
};

CPSPageGrid.propTypes = {
  children: node.isRequired,
};

PopOutGridItemMedium.propTypes = {
  children: node.isRequired,
  gridColumnStart: number,
  gridSpan: number,
};

PopOutGridItemMedium.defaultProps = {
  gridColumnStart: 1,
  gridSpan: 4,
};

export default Grid;
