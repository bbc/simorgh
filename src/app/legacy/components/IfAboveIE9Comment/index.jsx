import React from 'react';
import { node } from 'prop-types';

/*
 * This stops any children from loading on IE9 and below.
 */

const IfAboveIE9 = ({ children }) => (
  <>
    {/* eslint-disable-next-line react/no-danger */}
    <div dangerouslySetInnerHTML={{ __html: '<!--[if !IE]><!-->' }} />
    {children}
    {/* eslint-disable-next-line react/no-danger */}
    <div dangerouslySetInnerHTML={{ __html: '<!--<![endif]-->' }} />
  </>
);

IfAboveIE9.propTypes = {
  children: node.isRequired,
};

export default IfAboveIE9;
