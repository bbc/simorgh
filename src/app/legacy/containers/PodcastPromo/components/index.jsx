import React from 'react';
import styled from '@emotion/styled';
import { string, shape, arrayOf, element } from 'prop-types';
import { scriptPropType } from '#psammead/gel-foundations/src/prop-types';
import { GEL_SPACING_DBL } from '#psammead/gel-foundations/src/spacings';

import Title from './title';
import Card from './card';
import CardContent from './card-content';
import CardLink from './card-link';
import CardImageWrapper from './card-image-wrapper';
import CardTitle from './card-title';
import CardDescription from './card-description';
import CardEpisodesText from './card-episodes-text';

const PodcastContext = React.createContext({});
const withPodcastContext = Component => props =>
  (
    <PodcastContext.Consumer>
      {context => <Component {...context} {...props} />}
    </PodcastContext.Consumer>
  );

const Wrapper = styled.section`
  background-color: ${props => props.theme.palette.LUNAR};
  padding: ${GEL_SPACING_DBL};
`;

const PodcastPromo = ({ script, service, children, ...props }) => (
  <PodcastContext.Provider value={{ script, service }}>
    <Wrapper {...props}>{children}</Wrapper>
  </PodcastContext.Provider>
);

PodcastPromo.Title = withPodcastContext(Title);
PodcastPromo.Card = Card;
PodcastPromo.Card.Link = CardLink;
PodcastPromo.Card.ImageWrapper = CardImageWrapper;
PodcastPromo.Card.Content = CardContent;
PodcastPromo.Card.Title = withPodcastContext(CardTitle);
PodcastPromo.Card.Description = withPodcastContext(CardDescription);
PodcastPromo.Card.EpisodesText = withPodcastContext(CardEpisodesText);

PodcastPromo.propTypes = {
  children: arrayOf(element).isRequired,
  script: shape(scriptPropType).isRequired,
  service: string.isRequired,
};

PodcastPromo.defaultProps = {};

export default PodcastPromo;
