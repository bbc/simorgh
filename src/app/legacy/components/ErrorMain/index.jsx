import React from 'react';
import { string, arrayOf, shape } from 'prop-types';
import styled from '@emotion/styled';
import InlineLink from '#psammead/psammead-inline-link/src';
import Paragraph from '#psammead/psammead-paragraph/src';
import { getSerifMedium } from '#psammead/psammead-styles/src/font-styles';
import {
  getCanon,
  getParagon,
  GEL_FF_REITH_SANS,
} from '#psammead/gel-foundations/src/typography';
import idSanitiser from '#lib/utilities/idSanitiser';
import Grid, { GelPageGrid } from '#components/Grid';
import { scriptPropType } from '#psammead/gel-foundations/src/prop-types';

const StatusCode = styled.span`
  ${props => (props.script ? getParagon(props.script) : '')}
  color: ${props => props.theme.palette.POSTBOX};
  display: block;
  font-family: ${GEL_FF_REITH_SANS};
  font-weight: 600;
  padding: 2.5rem 0 0.5rem 0;
`;

const Heading = styled.h1`
  ${({ script }) => script && getCanon(script)}
  ${({ service }) => getSerifMedium(service)}
  color: ${props => props.theme.palette.SHADOW};
  margin-top: 0;
`;

const StyledGelPageGrid = styled(GelPageGrid)`
  padding-bottom: 4rem;
`;

const CustomParagraph = styled(Paragraph)`
  padding-top: 0.2rem;
`;

const ErrorMain = ({
  statusCode,
  title,
  message,
  solutions,
  callToActionFirst,
  callToActionLinkText,
  callToActionLinkUrl,
  callToActionLast,
  script,
  service,
}) => (
  <StyledGelPageGrid
    as="main"
    role="main"
    columns={{
      group0: 6,
      group1: 6,
      group2: 6,
      group3: 6,
      group4: 8,
      group5: 20,
    }}
    enableGelGutters
  >
    <Grid
      item
      startOffset={{
        group0: 1,
        group1: 1,
        group2: 1,
        group3: 1,
        group4: 2,
        group5: 5,
      }}
      columns={{
        group0: 6,
        group1: 6,
        group2: 6,
        group3: 6,
        group4: 6,
        group5: 12,
      }}
      margins={{ group0: true, group1: true, group2: true, group3: true }}
    >
      <StatusCode script={script} data-e2e="status-code">
        {statusCode}
      </StatusCode>
      <Heading id="content" script={script} service={service} tabIndex="-1">
        {title}
      </Heading>
      <CustomParagraph script={script} service={service}>
        {message}
      </CustomParagraph>
      <ul>
        {solutions.map(text => (
          <CustomParagraph
            script={script}
            service={service}
            as="li"
            key={idSanitiser(text)}
          >
            {text}
          </CustomParagraph>
        ))}
      </ul>
      <CustomParagraph script={script} service={service}>
        {callToActionFirst}
        <InlineLink
          href={callToActionLinkUrl}
          className="focusIndicatorReducedWidth"
        >
          {callToActionLinkText}
        </InlineLink>
        {callToActionLast}
      </CustomParagraph>
    </Grid>
  </StyledGelPageGrid>
);

ErrorMain.propTypes = {
  statusCode: string.isRequired,
  title: string.isRequired,
  message: string.isRequired,
  solutions: arrayOf(string).isRequired,
  callToActionFirst: string,
  callToActionLinkText: string.isRequired,
  callToActionLinkUrl: string.isRequired,
  callToActionLast: string,
  script: shape(scriptPropType).isRequired,
  service: string.isRequired,
};

ErrorMain.defaultProps = {
  callToActionFirst: null,
  callToActionLast: null,
};

export default ErrorMain;
