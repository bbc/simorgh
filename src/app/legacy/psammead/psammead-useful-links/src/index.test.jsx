import React from 'react';
import { render } from '../../../../components/react-testing-library-with-providers';
import latin from '../../../../components/ThemeProvider/fontScripts/latin';
import { UsefulLink, UsefulLinksLi, UsefulLinksUl } from './index';

const usefulCaptions = [
  {
    name: 'Mitocinmu da sauko da sautin labarai',
    url: 'https://www.bbc.com/igbo/afirika-49883577',
  },
  {
    name: 'Labaran BBC Hausa a text',
    url: 'https://www.bbc.com/igbo/afirika-49872694',
  },
  {
    name: 'Abokan huldar BBC Hausa',
    url: 'https://www.bbc.com/igbo/afirika-49869003',
  },
  {
    name: 'Timi Frank: Osinbajo ya maka mutum biyu',
    url: 'https://www.bbc.com/igbo/afirika-49883189',
  },
  {
    name: 'Gwaninta ba ta karbi wani dan Nijeriya',
    url: 'https://www.bbc.com/igbo/afirika-49869001',
  },
];

describe('One useful link', () => {
  it('should render correctly', () => {
    const { container } = render(
      <UsefulLink script={latin} service="news" href={usefulCaptions[0].url}>
        {usefulCaptions[0].name}
      </UsefulLink>,
    );
    expect(container).toMatchSnapshot();
  });
});

describe('Multiple useful links', () => {
  it('should render correctly', () => {
    const { container } = render(
      <UsefulLinksUl>
        {usefulCaptions.map(item => {
          return (
            <UsefulLinksLi key={usefulCaptions.indexOf(item)}>
              <UsefulLink script={latin} service="news" href={item.url}>
                {item.name}
              </UsefulLink>
            </UsefulLinksLi>
          );
        })}
      </UsefulLinksUl>,
    );
    expect(container).toMatchSnapshot();
  });
});
