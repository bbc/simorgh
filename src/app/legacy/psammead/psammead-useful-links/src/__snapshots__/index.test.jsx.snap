// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Multiple useful links should render correctly 1`] = `
.emotion-0 {
  padding: 0;
  margin: 0;
  list-style-type: none;
}

@media (min-width: 37.5rem) {
  .emotion-0 {
    -webkit-column-count: 2;
    column-count: 2;
    -webkit-column-gap: 1rem;
    column-gap: 1rem;
  }

  @supports (grid-template-columns: fit-content(200px)) {
    .emotion-0 {
      display: grid;
      grid-auto-flow: column;
      grid-template-columns: repeat(2, 1fr);
      grid-template-rows: repeat(
              3,
              auto
            );
      grid-column-gap: 1rem;
    }
  }
}

.emotion-2 {
  padding-top: 0.5rem;
  width: 100%;
}

@media (max-width: 37.4375rem) {
  .emotion-2:first-child {
    padding-top: 0;
  }
}

@media (min-width: 37.5rem) {
  .emotion-2 {
    display: inline-block;
  }

  @supports (grid-template-columns: fit-content(200px)) {
    .emotion-2 {
      display: block;
      -webkit-align-self: start;
      -ms-flex-item-align: start;
      align-self: start;
      width: initial;
    }
  }
}

.emotion-4 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-4 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

.emotion-4:hover,
.emotion-4:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-4:visited {
  color: #6E6E73;
}

<div>
  <ul
    class="emotion-0 emotion-1"
    role="list"
  >
    <li
      class="emotion-2 emotion-3"
      role="listitem"
    >
      <a
        class="emotion-4 emotion-5"
        href="https://www.bbc.com/igbo/afirika-49883577"
      >
        Mitocinmu da sauko da sautin labarai
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
      role="listitem"
    >
      <a
        class="emotion-4 emotion-5"
        href="https://www.bbc.com/igbo/afirika-49872694"
      >
        Labaran BBC Hausa a text
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
      role="listitem"
    >
      <a
        class="emotion-4 emotion-5"
        href="https://www.bbc.com/igbo/afirika-49869003"
      >
        Abokan huldar BBC Hausa
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
      role="listitem"
    >
      <a
        class="emotion-4 emotion-5"
        href="https://www.bbc.com/igbo/afirika-49883189"
      >
        Timi Frank: Osinbajo ya maka mutum biyu
      </a>
    </li>
    <li
      class="emotion-2 emotion-3"
      role="listitem"
    >
      <a
        class="emotion-4 emotion-5"
        href="https://www.bbc.com/igbo/afirika-49869001"
      >
        Gwaninta ba ta karbi wani dan Nijeriya
      </a>
    </li>
  </ul>
</div>
`;

exports[`One useful link should render correctly 1`] = `
.emotion-0 {
  font-size: 0.9375rem;
  line-height: 1.25rem;
  font-family: ReithSerif,Helvetica,Arial,sans-serif;
  font-weight: 500;
  font-style: normal;
  color: #222222;
  -webkit-text-decoration: none;
  text-decoration: none;
}

@media (min-width: 20rem) and (max-width: 37.4375rem) {
  .emotion-0 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

@media (min-width: 37.5rem) {
  .emotion-0 {
    font-size: 1rem;
    line-height: 1.25rem;
  }
}

.emotion-0:hover,
.emotion-0:focus {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.emotion-0:visited {
  color: #6E6E73;
}

<div>
  <a
    class="emotion-0 emotion-1"
    href="https://www.bbc.com/igbo/afirika-49883577"
  >
    Mitocinmu da sauko da sautin labarai
  </a>
</div>
`;
